src/query/service/src/interpreters/interpreter_catalog_create.rs

```diff
@@ -59,6 +63,25 @@ impl Interpreter for CreateCatalogInterpreter {
         }
 
         let catalog_manager = CatalogManager::instance();
+
+        // Build and check if catalog is valid.
+        let ctl = catalog_manager
+            .build_catalog(&CatalogInfo {
+                id: CatalogId::default(),
+                name_ident: CatalogNameIdent {
+                    tenant: self.plan.tenant.clone(),
+                    catalog_name: self.plan.catalog.clone(),
+                },
+                meta: CatalogMeta {
+                    catalog_option: self.plan.meta.catalog_option.clone(),
+                    created_on: chrono::Utc::now(),
+                },
+            })
+            .map_err(|err| err.add_message("The catalog you're trying to create is invalid."))?;
```

Comment:
The catalog you're trying to create is invalid.-> Error creating catalog.

Reply:
addressed.